client/webserver: prefer site in cwd, improve error

chappjc opened the pull request:
Have client/webserver.WebServer look for the "site" directory in the executable directory, then current working directory, then the repo-relative path.  Supplement the error message when site is not found with more information about where the program looked and where the user should put it.

chappjc commented:
I'm counting on your websocket/rpcserver/webserver PR to go in first... the panic is a bug in dexc main setup error handling.

buck54321 commented on client/webserver/webserver_test.go:
```diff
@@ -315,6 +317,29 @@ func TestMain(m *testing.M) {
+
+	c := &TCore{}
+	_, err = New(c, "127.0.0.1:0", tLogger, false)
+	if err == nil && !strings.HasPrefix(err.Error(), "no HTML template files found") {
```
~~`&&`~~ ➡ `||`

buck54321 commented on client/webserver/webserver_test.go:
```diff
@@ -315,6 +317,29 @@ func TestMain(m *testing.M) {
+	if err == nil && !strings.HasPrefix(err.Error(), "no HTML template files found") {
+		t.Errorf("Should have failed to start with no site folder.")
+	}
+	t.Log(err)
```
If intentional, please add some context to the error.

chappjc commented:
Sorry to change this so drastically.  After the [splitws PR](https://github.com/decred/dcrdex/pull/585) went in, I addressed the dexc main shutdown error handling.  Also changed the webserver to look in the **executable's** path rather than the working directory.  This matches the new wording in https://github.com/decred/dcrdex/pull/603/files#diff-04c6e90faac2675aa89e2176d2eec7d8R187-R191.

chappjc commented:
Ugh, damn that breaks test that run in folders like "/tmp/go-build384681446/b001"

chappjc commented on client/webserver/webserver.go:
```diff
@@ -105,27 +105,50 @@ func New(core clientCore, addr string, logger dex.Logger, reloadHTML bool) (*Web
+	}
+	execPath = filepath.Dir(execPath) // e.g. /opt/decred/dex
+
+	// executable path
+	siteDir := filepath.Join(execPath, "site") // e.g. /opt/decred/dex/site
+	log.Debugf("Looking for site in %v", siteDir)
+	if !folderExists(siteDir) {
+
+		// working directory
+		siteDir, _ = filepath.Abs("site")
+		log.Debugf("Looking for site in %v", siteDir)
+		if !folderExists(siteDir) {
+
+			// repo
+			siteDir = filepath.Clean(filepath.Join(execPath, "../../webserver/site"))
+			log.Debugf("Looking for site in %v", siteDir)
+			if !folderExists(siteDir) {
+
+				return nil, fmt.Errorf("no HTML template files found. "+
+					"Place the 'site' folder in the executable's directory %q or the working directory, "+
+					"or run dexc from within the client/cmd/dexc source workspace folder.", execPath)
```
Amazingly, a switch was much uglier, and running all the filepath ops ahead of time is a waste.
